staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -911,7 +911,13 @@ func (c completedConfig) New(name string, delegationTarget DelegationTarget) (*G
 	if s.isPostStartHookRegistered(priorityAndFairnessConfigConsumerHookName) {
 	} else if c.FlowControl != nil {
 		err := s.AddPostStartHook(priorityAndFairnessConfigConsumerHookName, func(hookContext PostStartHookContext) error {
-			go c.FlowControl.Run(hookContext.Done())
+			go func() {
+				if err := c.FlowControl.Start(hookContext); err != nil {
+					klog.FromContext(hookContext).Error(err, "Failed starting flow control")
+					// nothing more we can do here.
+				}
+			}()
+
```

Comment:
Wouldn't it be simpler to just keep `Run` and make it take the `hookContext` as the parameter?

Reply:
Maybe. Lacking the time to turn everything around again. Follow-up is fine, or takeover over of the PR too.

Reply:
See https://github.com/kubernetes/kubernetes/pull/129032#pullrequestreview-2558160845